app/renderer/utils/locator-generation.js

```diff
@@ -70,16 +78,24 @@ export function getSimpleSuggestedLocators(attributes, sourceDoc, isNative = tru
  *
  * @param {string} path a dot-separated string of indices
  * @param {Document} sourceDoc
+ * @param {boolean} isNative whether native context is active
+ * @param {string} automationName
  * @returns {Object.<string, string>} mapping of strategies to selectors
  */
-export function getComplexSuggestedLocators(path, sourceDoc) {
+export function getComplexSuggestedLocators(path, sourceDoc, isNative, automationName) {
   let complexLocators = {};
   const domNode = findDOMNodeByPath(path, sourceDoc);
   if (domNode.tagName.includes('XCUIElement')) {
     // XCUI native context
     const optimalClassChain = getOptimalClassChain(sourceDoc, domNode);
     complexLocators['-ios class chain'] = optimalClassChain ? '**' + optimalClassChain : null;
     complexLocators['-ios predicate string'] = getOptimalPredicateString(sourceDoc, domNode);
+  } else if (automationName.toLowerCase() === 'uiautomator2' && isNative) {
```

Comment:
[nits]
Could refer to https://github.com/appium/appium-inspector/blob/0405a47f764c5a3d637ac6dcf105fb1da6956c3e/app/renderer/components/Inspector/shared.js#L90 to avoid using writing `uiautomator2` directly? Possbly the shared components need to move to another place?

Reply:
I agree with splitting/moving `shared.js` elsewhere, but this would involve other files unrelated to this PR, so I would like to do that in a separate PR.